[BUG] MCP Server Jetbrains plugin

## Bug Description

I have installed the MCP Server Jetbrains plugin in Android Studio Quail 1

https://www.jetbrains.com/help/idea/mcp-server.html

I have enabled the MCP server in the settings.

In DevoxxGenie plugin I have tried adding this MCP server via HTTP_SSE

```
{
  "type": "sse",
  "url": "http://127.0.0.1:64342/sse",
  "headers": {}
}
```

When clicking on Test Connection it always fails with a HTTP405 error:

`Failed to connect to MCP server

Root cause: Unexpected status code: 405

Stack trace:
java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.RuntimeException: Unexpected status code: 405
	at dev.langchain4j.mcp.client.DefaultMcpClient.initialize(DefaultMcpClient.java:233)
	at dev.langchain4j.mcp.client.DefaultMcpClient.<init>(DefaultMcpClient.java:206)
	at dev.langchain4j.mcp.client.DefaultMcpClient$Builder.build(DefaultMcpClient.java:1103)
	at com.devoxx.genie.ui.settings.mcp.dialog.transport.HttpSseTransportPanel.createClient(HttpSseTransportPanel.java:128)
	at com.devoxx.genie.ui.settings.mcp.dialog.transport.TransportPanel.createClient(TransportPanel.java:89)
	at com.devoxx.genie.ui.settings.mcp.dialog.MCPServerDialog.lambda$testConnection$5(MCPServerDialog.java:307)
	at com.intellij.openapi.progress.impl.CoreProgressManager$2.run(CoreProgressManager.java:411)
	at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:561)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:197)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressSynchronously$10(CoreProgressManager.java:651)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$new$0(ProgressRunner.java:105)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:284)
	at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:98)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$1(CoreProgressManager.java:287)
	at com.intellij.platform.diagnostic.telemetry.helpers.TraceKt.use(trace.kt:44)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:286)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$14(CoreProgressManager.java:744)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:819)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:775)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:743)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:87)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:267)
	at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:98)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$5(ProgressRunner.java:284)
	at com.intellij.openapi.progress.impl.ProgressRunner$ProgressRunnable.run(ProgressRunner.java:556)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$launchTask$17(ProgressRunner.java:511)
	at com.intellij.concurrency.ThreadContext.installThreadContext(threadContext.kt:305)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$launchTask$18(ProgressRunner.java:510)
	at com.intellij.util.concurrency.ChildContext$runInChildContext$1.invoke(propagation.kt:192)
	at com.intellij.util.concurrency.ChildContext$runInChildContext$1.invoke(propagation.kt:192)
	at com.intellij.util.concurrency.ChildContext.runInChildContext(propagation.kt:198)
	at com.intellij.util.concurrency.ChildContext.runInChildContext(propagation.kt:192)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$launchTask$19(ProgressRunner.java:507)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Unknown Source)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.ExecutionException: java.lang.RuntimeException: Unexpected status code: 405
	at java.base/java.util.concurrent.CompletableFuture.reportGet(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.get(Unknown Source)
	at dev.langchain4j.mcp.client.DefaultMcpClient.initialize(DefaultMcpClient.java:228)
	... 39 more
Caused by: java.lang.RuntimeException: Unexpected status code: 405
	at dev.langchain4j.mcp.client.transport.http.StreamableHttpMcpTransport.lambda$execute$0(StreamableHttpMcpTransport.java:232)
	at java.net.http/jdk.internal.net.http.Http1Exchange.createResponseSubscriber(Unknown Source)
	at java.net.http/jdk.internal.net.http.Http1Exchange.readBodyAsync(Unknown Source)
	at java.net.http/jdk.internal.net.http.Exchange.readBodyAsync(Unknown Source)
	at java.net.http/jdk.internal.net.http.MultiExchange.lambda$responseAsync0$4(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.postFire(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$Completion.run(Unknown Source)
	at java.net.http/jdk.internal.net.http.HttpClientImpl$DelegatingExecutor.execute(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$UniCompletion.claim(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.complete(Unknown Source)
	at java.net.http/jdk.internal.net.http.Http1Response$HeadersReader.handle(Unknown Source)
	at java.net.http/jdk.internal.net.http.Http1Response$HeadersReader.handle(Unknown Source)
	at java.net.http/jdk.internal.net.http.Http1Response$Receiver.accept(Unknown Source)
	at java.net.http/jdk.internal.net.http.Http1Response$HeadersReader.tryAsyncReceive(Unknown Source)
	at java.net.http/jdk.internal.net.http.Http1AsyncReceiver.flush(Unknown Source)
	at java.net.http/jdk.internal.net.http.common.SequentialScheduler$LockingRestartableTask.run(Unknown Source)
	at java.net.http/jdk.internal.net.http.common.SequentialScheduler$CompleteRestartableTask.run(Unknown Source)
	at java.net.http/jdk.internal.net.http.common.SequentialScheduler$SchedulableTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	... 1 more
`

Accessing the url via a browser returns a 200 OK. What is the exact request the plugin is constructing? Maybe with a curl I would be able to view more HTTP response logging.

## Expected Behavior

MCP Server can be used

## Environment

- Android Studio Quail 1
- DevoxxGenie 1.8.8
- Windows 11

--- a/src/main/java/com/devoxx/genie/service/mcp/MCPExecutionService.java
+++ b/src/main/java/com/devoxx/genie/service/mcp/MCPExecutionService.java
@@ -14,6 +14,7 @@ import dev.langchain4j.mcp.McpToolProvider;
 import dev.langchain4j.mcp.client.DefaultMcpClient;
 import dev.langchain4j.mcp.client.McpClient;
 import dev.langchain4j.mcp.client.transport.McpTransport;
+import dev.langchain4j.mcp.client.transport.http.HttpMcpTransport;
 import dev.langchain4j.mcp.client.transport.http.StreamableHttpMcpTransport;
 import dev.langchain4j.mcp.client.transport.stdio.StdioMcpTransport;
 import dev.langchain4j.service.tool.ToolProvider;
@@ -270,6 +271,7 @@ public class MCPExecutionService implements Disposable {
      * @return An initialized MCP client or null if creation fails
      */
     @Nullable
+    @SuppressWarnings({"deprecation", "removal"}) // SSE transport is required for SSE-only servers (issue #1151)
     static McpClient initHttpSseClient(@NotNull MCPServer mcpServer) {
         try {
             String sseUrl = mcpServer.getUrl();
@@ -279,11 +281,13 @@ public class MCPExecutionService implements Disposable {
                 return null;
             }
 
-            MCPService.logDebug("Initializing streamable HTTP transport for HTTP_SSE config with URL: " + sseUrl);
+            MCPService.logDebug("Initializing SSE transport for HTTP_SSE config with URL: " + sseUrl);
 
-            // Use Streamable HTTP transport as replacement for legacy HTTP/SSE transport
-            StreamableHttpMcpTransport.Builder transportBuilder = new StreamableHttpMcpTransport.Builder()
-                    .url(sseUrl)
+            // Use the SSE-based transport for HTTP_SSE servers. The streamable HTTP transport POSTs
+            // to the single URL, which SSE-only endpoints (e.g. the JetBrains MCP server) reject with
+            // HTTP 405 (see issue #1151). HttpMcpTransport opens the event stream via a GET on /sse.
+            HttpMcpTransport.Builder transportBuilder = new HttpMcpTransport.Builder()
+                    .sseUrl(sseUrl)
                     .timeout(java.time.Duration.ofSeconds(DevoxxGenieStateService.getInstance().getTimeout()))
                     .logRequests(MCPService.isDebugLogsEnabled())
                     .logResponses(MCPService.isDebugLogsEnabled())

--- a/src/main/java/com/devoxx/genie/ui/settings/mcp/dialog/transport/HttpSseTransportPanel.java
+++ b/src/main/java/com/devoxx/genie/ui/settings/mcp/dialog/transport/HttpSseTransportPanel.java
@@ -17,7 +17,7 @@ import lombok.extern.slf4j.Slf4j;
 import dev.langchain4j.mcp.client.DefaultMcpClient;
 import dev.langchain4j.mcp.client.McpClient;
 import dev.langchain4j.mcp.client.transport.McpTransport;
-import dev.langchain4j.mcp.client.transport.http.StreamableHttpMcpTransport;
+import dev.langchain4j.mcp.client.transport.http.HttpMcpTransport;
 
 /**
  * Panel for configuring HTTP SSE MCP transport
@@ -94,6 +94,29 @@ public class HttpSseTransportPanel implements TransportPanel {
 
     @Override
     public McpClient createClient(Map<String, String> headers) throws Exception {
+        McpTransport transport = createTransport(headers);
+
+        // Create and return the client
+        return new DefaultMcpClient.Builder()
+                .clientName("DevoxxGenie")
+                .protocolVersion("2024-11-05")
+                .transport(transport)
+                .build();
+    }
+
+    /**
+     * Builds the MCP transport for this panel without opening a connection.
+     * <p>
+     * This uses the SSE-based {@link HttpMcpTransport} (which opens the event stream via a GET
+     * request to the configured {@code /sse} endpoint). The legacy SSE transport must NOT be
+     * replaced by {@code StreamableHttpMcpTransport}: the streamable transport POSTs to the single
+     * URL, and an SSE-only endpoint (e.g. the JetBrains MCP server) rejects that with HTTP 405
+     * (see issue #1151).
+     * <p>
+     * Package-private for testing.
+     */
+    @SuppressWarnings({"deprecation", "removal"}) // SSE transport is required for SSE-only servers (issue #1151)
+    McpTransport createTransport(Map<String, String> headers) {
         String sseUrl = sseUrlField.getText().trim();
 
         // Validate URL
@@ -107,9 +130,8 @@ public class HttpSseTransportPanel implements TransportPanel {
 
         log.debug("Creating HTTP SSE transport with URL: {}", sseUrl);
 
-        // Use Streamable HTTP transport (recommended replacement for legacy HTTP/SSE transport)
-        StreamableHttpMcpTransport.Builder transportBuilder = new StreamableHttpMcpTransport.Builder()
-                .url(sseUrl)
+        HttpMcpTransport.Builder transportBuilder = new HttpMcpTransport.Builder()
+                .sseUrl(sseUrl)
                 .timeout(Duration.ofSeconds(60))
                 .logRequests(true)
                 .logResponses(true);
@@ -118,14 +140,7 @@ public class HttpSseTransportPanel implements TransportPanel {
             transportBuilder.customHeaders(headers);
         }
 
-        McpTransport transport = transportBuilder.build();
-
-        // Create and return the client
-        return new DefaultMcpClient.Builder()
-                .clientName("DevoxxGenie")
-                .protocolVersion("2024-11-05")
-                .transport(transport)
-                .build();
+        return transportBuilder.build();
     }
 
     @Override

--- a/src/test/java/com/devoxx/genie/ui/settings/mcp/dialog/transport/HttpSseTransportPanelTest.java
+++ b/src/test/java/com/devoxx/genie/ui/settings/mcp/dialog/transport/HttpSseTransportPanelTest.java
@@ -1,0 +1,44 @@
+package com.devoxx.genie.ui.settings.mcp.dialog.transport;
+
+import dev.langchain4j.mcp.client.transport.McpTransport;
+import dev.langchain4j.mcp.client.transport.http.HttpMcpTransport;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import javax.swing.JTextField;
+import java.lang.reflect.Field;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Regression test for issue #1151.
+ * <p>
+ * The "HTTP SSE" transport panel must build an SSE-based {@link HttpMcpTransport}
+ * (which GETs the {@code /sse} endpoint to open the event stream), NOT a
+ * {@code StreamableHttpMcpTransport} (which POSTs to the single URL). Using the
+ * streamable transport against a legacy SSE endpoint such as the JetBrains MCP
+ * server's {@code /sse} endpoint results in an HTTP 405 (Method Not Allowed).
+ */
+class HttpSseTransportPanelTest {
+
+    private HttpSseTransportPanel panel;
+
+    @BeforeEach
+    void setUp() throws Exception {
+        panel = new HttpSseTransportPanel();
+
+        Field urlField = HttpSseTransportPanel.class.getDeclaredField("sseUrlField");
+        urlField.setAccessible(true);
+        ((JTextField) urlField.get(panel)).setText("http://127.0.0.1:64342/sse");
+    }
+
+    @Test
+    void createTransportUsesSseTransport() {
+        McpTransport transport = panel.createTransport(Map.of());
+
+        assertThat(transport)
+                .as("HTTP SSE panel must use the SSE-based HttpMcpTransport, not the streamable HTTP transport (issue #1151)")
+                .isInstanceOf(HttpMcpTransport.class);
+    }
+}

